src/VirtualClient/VirtualClient.Actions/MongoDB/MongoDBExecutor.cs

```diff
@@ -0,0 +1,315 @@
+
+            try
+            {
+                string ycsbPath = this.PlatformSpecifics.Combine(this.YcsbPackagePath, "ycsb-0.5.0", "bin", "ycsb");
+                string javaExecutablePath = this.PlatformSpecifics.Combine(this.JDKPackagePath, "bin", "java");
+                string ycsbExecutablePath = this.PlatformSpecifics.Combine(this.YcsbPackagePath, "ycsb-0.5.0", "bin");
+
+                string convertToExeArgument = this.GetCommandLineArguments("convertExecutable");
+                string createDBArgument = this.GetCommandLineArguments("createDB");
+                string createLogArgument = this.GetCommandLineArguments("createLog");
+                string runMongoArgument = this.GetCommandLineArguments("runMongo");
+                string insertDataArgument = this.GetCommandLineArguments("insertData");
+                string updateDataArgument = this.GetCommandLineArguments("updateData");
+
+                this.SetEnvironmentVariable(EnvironmentVariable.JAVA_HOME, javaExecutablePath, EnvironmentVariableTarget.Process);
+
+                await this.ExecuteCommandAsync("chmod", convertToExeArgument, ycsbExecutablePath, telemetryContext, cancellationToken).ConfigureAwait(false);
+                await this.ExecuteCommandAsync("mkdir", createDBArgument, this.MongoDBPackagePath, telemetryContext, cancellationToken)
+                    .ConfigureAwait(false);
+                await this.ExecuteCommandAsync("mkdir", createLogArgument, this.MongoDBPackagePath, telemetryContext, cancellationToken)
+                .ConfigureAwait(false);
+                await this.ExecuteCommandAsync($"{this.DBPath}", runMongoArgument, this.MongoDBPackagePath, telemetryContext, cancellationToken)
+                           .ConfigureAwait(false);
+
+                /* ./ycsb load mongodb -s -P /home/azureuser/vc/content/linux-x64/packages/ycsb/ycsb-0.5.0/workloads/workloada -p recordcount=1000000 -threads 16
+                */
+                DateTime loadStartTime = DateTime.UtcNow;
+                var loadOutput = await this.ExecuteCommandAsync($"{ycsbPath}", insertDataArgument, this.YcsbPackagePath, telemetryContext, cancellationToken)
+                         .ConfigureAwait(false);
+                DateTime loadFinishTime = DateTime.UtcNow;
+
+                await this.CaptureMetricsAsync(loadOutput, insertDataArgument, loadStartTime, loadFinishTime, telemetryContext, cancellationToken)
+                    .ConfigureAwait(false);
+
+                /*./ycsb run mongodb -s -P /home/azureuser/vc/content/linux-x64/packages/ycsb/ycsb-0.5.0/workloads/workloada -p operationcount=1000000 -p recordcount=1000000 -threads 16
+                */
+                DateTime runStartTime = DateTime.UtcNow;
+                var runOutput = await this.ExecuteCommandAsync($"{ycsbPath}", updateDataArgument, this.YcsbPackagePath, telemetryContext, cancellationToken)
```

Comment:
**load** resembles creation of database and **run** resembles actual execution of transactions between client and server. In benchmarking MongoDB, do we collect information also from load function?